src/antigravity/memory.py

```
@@ -0,0 +1,259 @@
+"""
+Infinite Memory Engine
+
+Implements recursive summarization to manage context window limits.
+Provides persistent memory storage and retrieval.
+"""
+
+from typing import List, Dict, Any, Optional, Callable
```

Ruff (0.14.8)

Import from `collections.abc` instead: `Callable`

Import from `collections.abc`

(UP035)